First Task/src/main/java/task/Functional.java

```diff
@@ -0,0 +1,23 @@
+package task;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+class Functional {
+    static int sum(List<Integer> list) {
+        int sum = 0;
+        for (Integer elem : list)
+            sum += elem;
+        return sum;
+    }
+
+    static int max(List<Integer> list) throws NoSuchElementException {
+        if (list.isEmpty())
+            throw new NoSuchElementException("Can define max element because the list is empty!");
+        int max = list.get(0);
+        for (Integer elem : list)
+            if (elem > max)
+                max = elem;
```

Comment:
```java
for (Integer elem : list) {
    if (elem > max) {
        max = elem;
    }
}
```
Yeah, I agree, it is a lot of boilerplate, but it is because of the language. However, using explicitly defined body it is easier to avoid problems working with this code in the future.